### ISSUE TITLE
Warning when installing and running under Jupyter 4

### ISSUE THREAD

**Issue by ajmazurie** [takluyver/bash_kernel#31]
Here is the warning message obtained during installation:

```
$ pip install bash_kernel
...
Successfully installed bash-kernel-0.4
$ python -m bash_kernel.install
/Users/ajmazurie/.pyenv/versions/_jupyter/lib/python2.7/site-packages/IPython/kernel/__init__.py:13: ShimWarning: The `IPython.kernel` package has been deprecated. You should import from ipykernel or jupyter_client instead.
  "You should import from ipykernel or jupyter_client instead.", ShimWarning)
Installing IPython kernel spec
```

And here is the message obtained when launching a notebook with the Bash kernel:

```
/Users/ajmazurie/.pyenv/versions/_jupyter/lib/python2.7/site-packages/IPython/kernel/__init__.py:13: ShimWarning: The `IPython.kernel` package has been deprecated. You should import from ipykernel or jupyter_client instead.
  "You should import from ipykernel or jupyter_client instead.", ShimWarning)
[IPKernelApp] ERROR | UNKNOWN MESSAGE TYPE: u'comm_open'
```

As far as I can tell the kernel seems to work well, however.

Thanks,
Aurelien

**Comment by takluyver** [comment]
Thanks. It shouldn't affect the working of the kernel, but we'll need to update the imports before ipython 5 comes out.

**Comment by mjbright** [comment]
I've been trying to install this module in the jupyter/minimal-notebook Docker image.

I get a similar message to the above when installing the bash_kernel.
> /opt/conda/lib/python3.4/site-packages/IPython/kernel/__init__.py:13: ShimWarning: The `IPython.kernel` package has been deprecated. You should import from ipykernel or jupyter_client instead.
>  "You should import from ipykernel or jupyter_client instead.", ShimWarning)
> Installing IPython kernel spec

In my case bash is not available in the "New" dropdown menu.

I'm confused as to how I can invoke this kernel on the command-line, I tried several things but none of these recognize the --kernel option.
>  ipython console --kernel bash
>  ipython notebook --kernel bash
>  jupyter notebook --kernel bash
  
Versions:
> ipython --version: 4.0.0
> jupyter --version: 4.0.4

I also tried modifying the install.py to import ipykernel or jupyter_client but got import errors.

Any suggestion?
This is my favourite kernel ;-)

**Comment by takluyver** [comment]
The console should recognise the `--kernel` option; the notebook won't, because you use the dropdown menu to select kernels instead.

It sounds like the kernel spec is not getting installed correctly. Can you see a `kernels/bash/ ` subfolder under any of the data directories? Run `jupyter --paths` to see where these are.

**Comment by mjbright** [comment]
Hi,

Yes I see that the bash kernel.json was being installed
under /root/.local/share/jupyter/kernels/bash/kernel.json ... not good.

So modifying my script to install with
    sudo -u jovyan python -m bash_kernel.install

did the trick.

Now I just have to automate that ... will check with docker-stacks for that
...

Thanks.

**Comment by takluyver** [comment]
I should probably make an option to do a systemwide install of the kernelspec rather than a per-user install. Maybe it should even do that automatically when run as root, but I'm not sure about that.

The code is already there, it's just not exposed at the command line. If you want to work on it, go for it.

**Comment by takluyver** [comment]
Closed by #39

**Comment by nico972** [comment]
Update the anaconda version by: 
>> conda update conda
And then 
>> conda update anaconda
